synapse/storage/databases/main/event_federation.py

```diff
@@ -936,15 +936,50 @@ def _get_backfill_events(self, txn, room_id, event_list, limit):
         # We want to make sure that we do a breadth-first, "depth" ordered
         # search.
 
+        # Look for the prev_event_id connected to the given event_id
         query = (
             "SELECT depth, prev_event_id FROM event_edges"
+            # Get the depth of the prev_event_id from the events table
             " INNER JOIN events"
             " ON prev_event_id = events.event_id"
+            # Find an event which matches the given event_id
             " WHERE event_edges.event_id = ?"
             " AND event_edges.is_state = ?"
             " LIMIT ?"
         )
 
+        # Look for the "insertion" events connected to the given event_id
+        # TODO: Do we need to worry about selecting only from the given room_id? The other query above doesn't
+        connected_insertion_event_query = (
+            "SELECT e.depth, i.event_id FROM insertion_event_edges AS i"
+            # Get the depth of the insertion event from the events table
+            " INNER JOIN events AS e"
+            " ON e.event_id = i.event_id"
```

Comment:
```suggestion
            " INNER JOIN events AS e USING (event_id)"
```

That way you don't need to specify the table when referecing `event_id`

Reply:
Updated ⏩ 

Is this the preferred way? To me, it's a bit more confusing to keep track of which `event_id` it is.